conans/client/tools/oss.py

```diff
@@ -24,14 +24,35 @@ def args_to_string(args):
         return " ".join("'" + arg.replace("'", r"'\''") + "'" for arg in args)
 
 
+class CpuProperties(object):
+
+    def get_cpu_quota(self):
+        return int(load("/sys/fs/cgroup/cpu/cpu.cfs_quota_us"))
+
+    def get_cpu_period(self):
+        return int(load("/sys/fs/cgroup/cpu/cpu.cfs_period_us"))
+
+    def get_cpus(self):
+        try:
+            cfs_quota_us = self.get_cpu_quota()
+            cfs_period_us = self.get_cpu_period()
+            if cfs_quota_us > 0 and cfs_period_us > 0:
+                return int(math.ceil(cfs_quota_us / cfs_period_us))
+        except:
+            pass
+        return multiprocessing.cpu_count()
+
+
 def cpu_count(output=None):
-    output = default_output(output, 'conans.client.tools.oss.cpu_count')
```

Comment:
Why removing this output? This would make the ``output.warn()`` below to crash if ``output=None`` (otherwise, remove the default and check with an assert for valid output object)